lodash/index.d.ts

```diff
@@ -11202,7 +11202,7 @@ declare namespace _ {
     }
 
     //_.cloneDeepWith
-    type CloneDeepWithCustomizer<TValue, TResult> = (value: TValue) => TResult;
+    type CloneDeepWithCustomizer<TValue, TResult> = (value: TValue, key?: number|string, object?: any, stack?: any) => TResult;
```

Comment:
The first internal call to the customizer doesn't include the 2nd, 3rd, or 4th arguments. Only the recursive calls to `baseClone` seem to include the remaining arguments.

We don't care about this when we're creating the customizers as library users, but shouldn't we take this into account for correctness? Or am I misinterpreting the use of optional parameters? Still a bit new to TypeScript.